samples/sample_hevc_fei_abr/include/task.h

```diff
@@ -21,38 +21,66 @@ or https://software.intel.com/en-us/media-client-solutions-support.
 #define __HEVC_FEI_ABR_TASK__
 
 #include "buffer_pool.h"
+#include "base_allocator.h"
+
 #include <queue>
 #include <cmath>
-
-//#define DEBUG_OUTPUT
-
-#ifdef DEBUG_OUTPUT
-#include <cstdio>
-#endif
-#include "base_allocator.h"
+#include <map>
 
 struct FrameStatData
 {
-    mfxU32 EncodedOrder   = 0xffffffff;
-    mfxU32 DisplayOrder   = 0xffffffff;
-    mfxU8  qp             = 0xff;
-    mfxF64 qstep0         = -1.0;
-    mfxF64 qstep          = -1.0;
-    mfxU16 frame_type     = MFX_FRAMETYPE_UNKNOWN;
-    mfxI32 poc            = -100;
-    //mfxU32 num_intra      = 0;
-    mfxF64 share_intra    = 0.0;
-    mfxF64 propagation    = 0.0;
-    //mfxU32 num_inter      = 0;
-    //mfxF64 share_inter    = 0.0;
-    //mfxU32 num_skip       = 0;
-    //mfxF64 share_skip     = 0.0;
-    mfxU32 frame_size     = 0;
-    //mfxF64 psnrY          = 0.0;
-    mfxF64 mseY           = 0.0;
-    mfxF64 complexity0    = 0.0;
-    mfxF64 complexity     = 0.0;
-    mfxF64 predicted_size = 0.0;
+    mfxU32 EncodedOrder                  = 0xffffffff;
+    mfxU32 DisplayOrder                  = 0xffffffff;
+    mfxU8  QP                            = 0xff;
+    mfxF64 QstepOriginal                 = -1.0;
+    mfxF64 QstepCalculated               = -1.0;
+    mfxU16 FrameType                     = MFX_FRAMETYPE_UNKNOWN;
+    mfxI32 POC                           = -100;
+    mfxF64 ShareIntra                    = 0.0;
+    mfxF64 Propagation                   = 0.0;
+    mfxU32 FrameSize                     = 0;
+    mfxF64 VisualDistortion              = 0.0; // MSE or any approximation like number of nonzero coefficients
+    mfxF64 ComplexityOriginal            = 0.0;
+    mfxF64 ComplexitySmoothed            = 0.0;
+
+    mfxU32 NPixelsPropagated             = 0;
+    mfxF64 NPixelsTransitivelyPropagated = 1.0;
+    mfxU32 NPixelsInFrame                = 0;
+
+    mfxU64 BitsEncoded                   = 0;
+    mfxU64 BitsPredicted                 = 0;
+
+    // Map of FrameDisplayOrder <=> NumOfPixels.
+    // Stores amount of pixels predicted FROM reference frame.
+    std::map<mfxU32, mfxU32> NumPixelsPredictedFromRef;
+    // Map of FrameDisplayOrder <=> ShareOfPixels.
+    // Shows share of INTER pixels predicted FROM reference frame.
+    std::map<mfxU32, mfxF64> ShareOfPredictedPixelsFromRef;
+
+    bool Contains(mfxU32 fo)
+    {
+        return NumPixelsPredictedFromRef.find(fo) != NumPixelsPredictedFromRef.end();
+    }
+
+    void AddPxlCount(mfxU32 fo, mfxU32 amount)
+    {
+        if (Contains(fo))
+        {
+            NumPixelsPredictedFromRef[fo] += amount;
+        }
+        else
+        {
+            NumPixelsPredictedFromRef[fo] = amount;
+        }
+    }
+
+    void FillProp()
+    {
+        for (auto & item : NumPixelsPredictedFromRef)
+        {
+            ShareOfPredictedPixelsFromRef[item.first] = mfxF64(item.second) / ((1.0 - ShareIntra) * NPixelsInFrame);
```

Comment:
Static analysis would complain about potential division by zero here.

Reply:
it's the only place where static analysis complains? will add a check